Chart with sensors is not displayed. Monitoring tools shows that the Chart SVG is updating, but SVG is not drawing

## Affected Version(s):
(The versions of fluidd this issue occurs on.)
- klipper version: v0.10.0-167-g2b7d0bba
- moonraker version: v0.7.1-194-geda5623
- fluidd version: v1.16.2

## Platform: Raspberry Pi 4 Model B Rev 1.4 on buster

## Hardware Configuration:
MCU v0.10.0-122-gc179db3d

## Reproduction Steps:
1. Installed the Mainsail
2. Changed in the Nginx the link for sites-enabled from fluidd to mainsail
3. Restore the link to fluidd (doesn't like mainsail)

## Actual Results:
The chart disappeared while the table part with sensors still exists and updating. The pressing CHART ON/OFF button is working but chart is not displayed while the container is appearing/disappearing. Have tried it in different browsers Edge,

## Expected Results:
The chart suppose to be displayed

## Log Files:
(The log files fluidd provides at Configure > System Control.)
- klipper log:
.....
'webhooks: registering remote method 'shutdown_machine' for connection id: 3041919864
webhooks: registering remote method 'reboot_machine' for connection id: 3041919864
webhooks: registering remote method 'pause_job_queue' for connection id: 3041919864
webhooks: registering remote method 'start_job_queue' for connection id: 3041919864
Stats 1133.8: gcodein=0 mcu: mcu_awake=0.000 mcu_task_avg=0.000000 mcu_task_stddev=0.000000 bytes_write=3856 bytes_read=4110 bytes_retransmit=0 bytes_invalid=0 send_seq=168 receive_seq=168 retransmit_seq=0 srtt=0.004 rttvar=0.002 rto=0.025 ready_bytes=0 stalled_bytes=0 freq=72004070  heater_bed: target=0 temp=0.0 pwm=0.000 motherboard_temp: temp=0.0 raspberry_pi: temp=48.2 sysload=0.12 cputime=27.165 memavail=3655984 print_time=0.001 buffer_time=0.000 print_stall=0 extruder: target=0 temp=0.0 pwm=0.000
Stats 1134.8: gcodein=0 mcu: mcu_awake=0.000 mcu_task_avg=0.000000 mcu_task_stddev=0.000000 bytes_write=3903 bytes_read=4286 bytes_retransmit=0 bytes_invalid=0 send_seq=172 receive_seq=172 retransmit_seq=0 srtt=0.003 rttvar=0.001 rto=0.025 ready_bytes=0 stalled_bytes=0 freq=72002110  heater_bed: target=0 temp=26.8 pwm=0.000 motherboard_temp: temp=28.9 raspberry_pi: temp=47.7 sysload=0.12 cputime=27.192 memavail=3657264 print_time=0.001 buffer_time=0.000 print_stall=0 extruder: target=0 temp=27.1 pwm=0.000`


- moonraker log:
...
'Klippy Host not connected
Traceback (most recent call last):
  File "/home/pi/moonraker/moonraker/websockets.py", line 224, in execute_method
    result = await method(conn, *args, **kwargs)
  File "/home/pi/moonraker/moonraker/websockets.py", line 317, in func
    user=ws.current_user))
  File "/home/pi/moonraker/moonraker/moonraker.py", line 546, in make_request
    return await self._request_standard(web_request)
  File "/home/pi/moonraker/moonraker/moonraker.py", line 602, in _request_standard
    return await base_request.wait()
  File "/home/pi/moonraker/moonraker/moonraker.py", line 784, in wait
    raise self.response
utils.ServerError: Klippy Host not connected
2021-11-29 15:22:57,073 [moonraker.py:connect()] - Klippy Connection Established
2021-11-29 15:22:57,631 [moonraker.py:_init_klippy_connection()] - Webhooks Subscribed
2021-11-29 15:22:57,633 [moonraker.py:_init_klippy_connection()] - GCode Output Subscribed
2021-11-29 15:22:58,906 [moonraker.py:_check_ready()] - Klippy ready
2021-11-29 15:22:59,389 [data_store.py:_init_sensors()] - Configuring available sensors: ['extruder']
2021-11-29 15:27:56,966 [moonraker.py:on_connection_closed()] - Klippy Connection Removed
2021-11-29 15:27:57,045 [websockets.py:build_error()] - JSON-RPC Request Error: 503
Klippy Host not connected
Traceback (most recent call last):
  File "/home/pi/moonraker/moonraker/websockets.py", line 224, in execute_method
    result = await method(conn, *args, **kwargs)
  File "/home/pi/moonraker/moonraker/websockets.py", line 317, in func
    user=ws.current_user))
  File "/home/pi/moonraker/moonraker/moonraker.py", line 546, in make_request
    return await self._request_standard(web_request)
  File "/home/pi/moonraker/moonraker/moonraker.py", line 602, in _request_standard
    return await base_request.wait()
  File "/home/pi/moonraker/moonraker/moonraker.py", line 784, in wait
    raise self.response
utils.ServerError: Klippy Host not connected
2021-11-29 15:27:58,227 [moonraker.py:connect()] - Klippy Connection Established
2021-11-29 15:27:58,792 [moonraker.py:_init_klippy_connection()] - Webhooks Subscribed
2021-11-29 15:27:58,794 [moonraker.py:_init_klippy_connection()] - GCode Output Subscribed
2021-11-29 15:28:00,055 [moonraker.py:_check_ready()] - Klippy ready
2021-11-29 15:28:00,556 [data_store.py:_init_sensors()] - Configuring available sensors: ['heater_bed', 'temperature_sensor motherboard_temp', 'temperature_sensor raspberry_pi', 'extruder']
2021-11-29 15:28:54,254 [websockets.py:remove_websocket()] - Websocket Removed: 2805303696
2021-11-29 15:28:55,013 [app.py:log_request()] - 101 GET /websocket?token=GX5AIIWGKJRTIBLEE5BKOVFY2OIUXQWP (192.168.1.13) [_TRUSTED_USER_] 3.49ms
2021-11-29 15:28:55,014 [websockets.py:add_websocket()] - New Websocket Added: 2805343888
2021-11-29 15:33:07,854 [websockets.py:remove_websocket()] - Websocket Removed: 2805343888
2021-11-29 15:34:20,873 [app.py:log_request()] - 101 GET /websocket?token=XSDCQ55FOBKFREJ562UMAY6LKMSRFVIT (192.168.1.13) [_TRUSTED_USER_] 3.33ms
2021-11-29 15:34:20,874 [websockets.py:add_websocket()] - New Websocket Added: 2805515760
2021-11-29 15:38:19,777 [websockets.py:remove_websocket()] - Websocket Removed: 2805515760
2021-11-29 15:41:28,392 [app.py:log_request()] - 101 GET /websocket?token=JAUBH6OFANDRTRCW4SVFYAB7YATO2CCN (192.168.1.13) [_TRUSTED_USER_] 3.89ms
2021-11-29 15:41:28,393 [websockets.py:add_websocket()] - New Websocket Added: 2805517712
2021-11-29 15:42:18,447 [websockets.py:remove_websocket()] - Websocket Removed: 2805517712
2021-11-29 15:42:19,959 [app.py:log_request()] - 101 GET /websocket?token=ZXF563PYDS4IK5UXDAQQU6EVVU2Z63K4 (192.168.1.13) [_TRUSTED_USER_] 5.33ms
2021-11-29 15:42:19,960 [websockets.py:add_websocket()] - New Websocket Added: 2804835472'

## Additional information
The debugging tools shows the SVG <g></g> is updating constantly

Sorry, have issue and too dumb to resolve..

```
pi@fluiddpi:~ $ git clone https://github.com/fluidd-core/fluidd.git
fatal: destination path 'fluidd' already exists and is not an empty directory.
```

As this already has fluidd inside.
Other commands do work, but my guess it that without this first part resolved I have error with following:

```
pi@fluiddpi:~/fluidd $ npm run serve
npm ERR! Missing script: "serve"
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
```

--- a/src/components/widgets/thermals/ThermalChart.vue
+++ b/src/components/widgets/thermals/ThermalChart.vue
@@ -204,8 +204,8 @@ export default class ThermalChart extends Vue {
         boundaryGap: false,
         max: 'dataMax',
         min: (value: any) => {
-          const serverConfig = this.$store.getters['server/getConfig']
-          return value.max - (serverConfig.server.temperature_store_size * 1000)
+          const temperature_store_size = this.$store.getters['charts/getChartRetention']
+          return value.max - (temperature_store_size * 1000)
         },
         axisTick: {
           show: false

--- a/src/store/charts/getters.ts
+++ b/src/store/charts/getters.ts
@@ -25,12 +25,9 @@ export const getters: GetterTree<ChartState, RootState> = {
    */
   getChartRetention: (state, getters, rootState, rootGetters) => {
     const config = rootGetters['server/getConfig']
-    return (
-      'server' in config &&
-      'temperature_store_size' in config.server
-    )
-      ? config.server.temperature_store_size
-      : Globals.CHART_HISTORY_RETENTION
+    return config.data_store?.temperature_store_size ??
+      config.server?.temperature_store_size ??
+      Globals.CHART_HISTORY_RETENTION
   },
 
   /**

--- a/src/store/printer/actions.ts
+++ b/src/store/printer/actions.ts
@@ -145,9 +145,11 @@ export const actions: ActionTree<PrinterState, RootState> = {
       }
 
       // Add a temp chart entry
-      const retention = (rootState.server)
-        ? rootState.server.config.server.temperature_store_size
-        : Globals.CHART_HISTORY_RETENTION
+      const rootStateServerConfig = rootState.server?.config
+      const retention =
+        rootStateServerConfig?.data_store?.temperature_store_size ??
+        rootStateServerConfig?.server?.temperature_store_size ??
+        Globals.CHART_HISTORY_RETENTION
       handleAddChartEntry(retention, getters.getChartableSensors)
     }
   }

--- a/src/store/server/index.ts
+++ b/src/store/server/index.ts
@@ -21,10 +21,7 @@ export const defaultState = (): ServerState => {
       authorization: {
         enabled: true
       },
-      server: {
-        gcode_store_size: 1000,
-        temperature_store_size: 1200
-      }
+      server: {}
     },
     moonraker_stats: [],
     throttled_state: {

--- a/src/store/server/types.ts
+++ b/src/store/server/types.ts
@@ -113,6 +113,6 @@ export interface ServerConfiguration {
 }
 
 export interface DataStoreConfiguration {
-  gcode_store_size: number;
-  temperature_store_size: number;
+  gcode_store_size?: number;
+  temperature_store_size?: number;
 }

--- a/src/store/server/types.ts
+++ b/src/store/server/types.ts
@@ -100,6 +100,7 @@ export interface DistroVersionParts {
 export interface ServerConfig {
   authorization: ServerAuthorization;
   server: ServerConfiguration;
+  data_store?: DataStoreConfiguration;
 }
 
 export interface ServerAuthorization {
@@ -107,6 +108,11 @@ export interface ServerAuthorization {
 }
 
 export interface ServerConfiguration {
+  gcode_store_size?: number;
+  temperature_store_size?: number;
+}
+
+export interface DataStoreConfiguration {
   gcode_store_size: number;
   temperature_store_size: number;
 }

